### PR TITLE
Nano codestyle GitHub action

### PR DESCRIPTION
**Pull request by yangw1234**
# Using github action to run quick checks like code style.

## A couple reasons to use github actions:
1. Github action is **fast** to pull code and install dependecies as they are running on azure .
2. The failing result is much more **eye-catching** than internal jenkins jobs. This will give developer the pressure to at least fix the all the failures in github action before merging the code.
3. Some **non-code changes can directly merge** if github actions passed with going through jenkins.
4. It can make sure the test is _actually run on **this** pr_. (We may input the wrong pr number in jenkins; We may also mistakenly use other pr's success result and think this pr is passed. All have happeded to me.)
4. It can **free up jenkins resources** to run large scale distributed tests.

## Security Issues:

Github add an option to require approvals for outside collaborators to run github actions.

image

## Future PRs will look like this after this PR merged.

image

(I intentionaly make a change to break code style check)

### CODE REVIEW

File: .github/workflows/nano-style-check.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install pycodestyle
+      
+      # Runs a single command using the runners shell
+      - name: Style checking
+        run: bash python/nano/test/run-nano-codestyle-test.sh
+        env:
+            ANALYTICS_ZOO_ROOT: ${{ github.workspace }}
```
Comment (jason-dai) [inline .github/workflows/nano-style-check.yml:37]: Why ANALYTICS_ZOO_ROOT?

Comment (yangw1234) [reply .github/workflows/nano-style-check.yml:37]: This environment variable is used in many script. 

You mean changing to BIGDL_ROOT?

Comment (jason-dai) [reply .github/workflows/nano-style-check.yml:37]: which scripts does this one use? And yes, changing to BIGDL_ROOT unless it's too much work

Comment (yangw1234) [reply .github/workflows/nano-style-check.yml:37]: > which scripts does this one use?

**This test runs**
```
line 35   run: bash python/nano/test/run-nano-codestyle-test.sh
```

**Scripts using ANALYTICS_ZOO_ROOT:**
```
./scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/example/run-scala-example-test-local.sh
./python/orca/dev/example/run-example-test-ray.sh
./python/orca/dev/example/run-example-test-ray-integration.sh
./python/nano/test/run-nano-test.sh
./python/nano/test/run-nano-test-ipex.sh
./python/nano/test/run-nano-type-test.sh
./python/nano/test/run-nano-codestyle-test.sh
./apps/run-scala-app-test.sh
./apps/run-app-tests-pytorch-pip.sh
./apps/run-app-tests-pip.sh
```

Comment (yangw1234) [reply .github/workflows/nano-style-check.yml:37]: > And yes, changing to BIGDL_ROOT unless it's too much work

Changing code is easy. Changing many jenkins job configs one by one could take some time. @jason-dai

**Comment by yangw1234** [comment]
I'll first merge this and change the environmental variable in another PR.

**Pull request merged**